bin/distmap

```diff
@@ -20,124 +20,166 @@ use DownloadTrimmedRead;
 
-my $verbose = 0;
-my $test = 0;
 my $help = 0;
+my $version = 0;
 
+## DEPRECATED (log warning if used?) - TODO: remove
+my $no_trim=0;
+my $qualThreshold=20;
+my $discardRemainingNs=0;
+my $minLength=40;
+my $trimQuality=0; ## TODO: be careful with this argument (it is really no-trim-quality switch)
+my $no5ptrim=0;
+my $verbose = 0;
 
+## OBSOLETE (log warning if used?) - TODO: remove
+my $picard_mergesamfiles_jar="";
+my $picard_sortsam_jar="";
+my $picard_mark_duplicates_jar=""; # TODO: is not even used - remove
+my $only_trim=0;
+my $trim_script_path="";
+my $fastqtype="illumina";
+my $nozip=0;
 
-#### Trimming parameters
+## UNDOCUMENTED - TODO: documment
+my $only_download_reads=0;
+my $mappers_exe_archive = "";
 
-    my $qualThreshold=20;
-    my $minLength=40;
-    my $fastqtype="illumina";
-    my $discardRemainingNs=0;
-    my $trimQuality=0; ## TODO: be careful with this argument (it is really no-trim-quality switch)
-    my $no5ptrim=0;
-    my $nozip=0;
+## TODO - supposedly options but not in the GetOptions block!!! cannot be specified
+## TODO - add to the parsing or remove support for them and default always!
+## TODO - they appear in the example!!
+my $job_priority="VERY_HIGH";
+my $hadoop_scheduler="Fair"; # this is not used at all (remove?)
 
+## UNUSED variable!! - TODO: remove?
+my $test = 0;
 
+## parse command line
 GetOptions(
-	"reference-fasta=s"			=>\$reference_fasta,	# to check
-	"input=s"                           	=> \@input_files,      # to check
-	"output=s"                          	=> \$output_directory,	# to check
-	"only-index" 				=> \$only_index,
-	"only-process" 				=> \$only_process,
-	"only-hdfs-upload" 			=> \$only_hdfs_upload,
-	"only-trim" 				=> \$only_trim,
-	"no-trim" 				=> \$no_trim,
+	# REQUIRED (to check)
+	"reference-fasta=s"				=>\$reference_fasta,
+	"output=s"						=> \$output_directory,
+	"input=s"						=> \@input_files,
+	"mapper=s"						=> \@mapper,
+	"mapper-path=s"					=> \@mapper_path,
+	"mapper-args=s"					=> \@mapper_args,
+
+	# PSEUDO-REQUIRED (to check in PATH)
+	"hadoop-home=s"					=>\$hadoop_home,
+	"picard=s"						=> \$picard,
+	"readtools=s"					=> \$readtools_arg,
+
+	# OPTIONAL
+	## pipeline
+	"only-index"					=> \$only_index,
+	"only-process"					=> \$only_process, # TODO - process/upload still different? (otherwise, deprecate)
+	"only-hdfs-upload"				=> \$only_hdfs_upload,
+	"only-map"						=> \$only_map,
+	"only-hdfs-download"			=> \$only_hdfs_download,
+	"only-merge"					=> \$only_merging, # TODO - should be deprecated?
+	"only-delete-temp"				=> \$only_delete_temp,
+	"trim-args"						=> \$trim_args,
+	## other
+	"tmp-dir=s"						=> \$tmp_dir,
+	"output-format=s"				=> \$output_format,
+	"reference-index-archive=s"		=> \$refindex_archive,
+	"queue-name=s"					=> \$queue_name,
+	"job-desc=s"					=> \$job_desc,
+	"gsnap-output-split"			=> \$gsnap_output_split, # TODO - this should be a conditional argument on GSNAP mapper (throw otherwise)
```

Comment:
Mapper arguments, like trimmer arguments, should ultimately be set in their args string and nowhere else. But maybe not in this round.

Reply:
I think that these two are special arguments: split requires a folder to output (if I remember correctly) and bwa-aln requires the sampe step later. But true, maybe for next round...

Reply:
Open #83 to track this